src/julia-syntax.scm

```diff
@@ -1984,6 +1984,31 @@
         ,@(if (length= e 3) '(()) '())
         ,@(map expand-forms (cddr e))))
 
+   'opaque_closure
+   (lambda (e)
+     (let* ((meth (caddr (caddr (expand-forms (cadr e))))) ;; `method` expr
+            (lam       (cadddr meth))
+            (sig-block (caddr meth))
+            (sig-block (if (and (pair? sig-block) (eq? (car sig-block) 'block))
+                           sig-block
+                           `(block ,sig-block)))
+            (stmts     (cdr (butlast sig-block)))
+            (sig-svec  (last sig-block))
+            (typ-svec  (caddr sig-svec))
+            (tvars     (cddr (cadddr sig-svec)))
+            (argtypes  (cdddr typ-svec))
+            (functionloc (cadr (caddddr sig-svec))))
+       (if (length= argtypes 0) `(_opaque_closure ,(expand-forms `(curly (core Tuple))) (false) 0 ,functionloc ,lam)
+         (let* ((vssa (make-ssavalue))
+                (vval (expand-forms (last argtypes)))
+                (argtypes (append (butlast argtypes) (list vssa)))
+                (argtype   (foldl (lambda (var ex) `(call (core UnionAll) ,var ,ex))
+                                  (expand-forms `(curly (core Tuple) ,@argtypes))
+                                  (reverse tvars))))
+            `(block
+               (= ,vssa ,vval)
+               (_opaque_closure ,argtype (call (core isa) ,vssa (core TypeofVararg)) ,(length argtypes) ,functionloc ,lam))))))
```

Comment:
I don't see a reason to pass the `isva` flag here, given that the front end doesn't use it.

Reply:
I need to pass it through to the code that creates `new_opaque_closure`. I'm not quite sure what you're suggesting I do instead.

Reply:
Have new_opaque_closure look at the last argument type itself?

Reply:
So keep the argtypes as a list here rather than creating the Tuple call here?